src/data/nat/factorial.lean

```diff
@@ -96,4 +96,22 @@ lemma self_le_factorial : ∀ n : ℕ, n ≤ n!
 | 0 := zero_le_one
 | (k+1) := le_mul_of_one_le_right k.zero_lt_succ.le (nat.one_le_of_lt $ nat.factorial_pos _)
 
+lemma lt_factorial_self {n : ℕ} (hi : 3 ≤ n) : n < n! :=
+begin
+  rw [← succ_pred_eq_of_pos ((zero_lt_two.trans (lt.base 2)).trans_le hi), factorial_succ],
+  exact lt_mul_of_one_lt_right ((pred n).succ_pos) ((one_lt_two.trans_le
+    (le_pred_of_lt (succ_le_iff.mp hi))).trans_le (self_le_factorial _)),
+end
+
+lemma add_factorial_lt_factorial_add {i : ℕ} (n : ℕ) (hi : 2 ≤ i) :
+  i + n.succ! < (i + n.succ)! :=
```

Comment:
I think we generally try to express `n.succ` as `(n+1)` where possible (because it looks nicer).

Also, should we have an alternate form `add_factorial_lt_factorial_add'` which has `1 ≤ n` as a hypothesis?

Reply:
I took care of the first, and will think about the second!

Reply:
I also added the lemma.